client/src/components/Attachment/UploadAttachment.js

```diff
@@ -25,26 +25,56 @@ const GQL_UPDATE_ATTACHMENT = gql`
   }
 `
 
-const UploadAttachment = ({ getRelatedObject, edit, saveRelatedObject }) => {
+const UploadAttachment = ({
+  edit,
+  saveRelatedObject,
+  relatedObjectType,
+  relatedObjectUuid
+}) => {
   const [error, setError] = useState(null)
   const [remove, setRemove] = useState(false)
   const [uploadedList, setUploadedList] = useState([])
-  const relatedObject = getRelatedObject()
+  const relatedObjectUuidRef = useRef(relatedObjectUuid)
 
-  const attachmentSave = async e => {
+  useEffect(() => {
+    if (relatedObjectUuid !== relatedObjectUuidRef.current) {
+      relatedObjectUuidRef.current = relatedObjectUuid
+    }
+  }, [relatedObjectUuidRef, relatedObjectUuid])
+
+  useEffect(() => {
+    // In the edit mode the base page entity (report, location, etc.) always has an uuid
+    // So when uploading new attachment it is always inserted related tables, (ie. see attachmentSave)
+    if (edit) {
+      return
+    }
+    // if base entity is not created, the attachment can not be associated
+    if (!relatedObjectUuid) {
+      return
+    }
+    // Update only object not associated with the base entity, as in report
+    const updateList = uploadedList.filter(
+      list => !list.attachmentRelatedObjects[0].relatedObjectUuid
+    )
+    for (let i = 0; i < updateList.length; i++) {
+      save(updateList[i], relatedObjectUuid, true)
```

Comment:
You could think in case of save is not succesfull. Like in attachment save.

Reply:
Yes error needs to be handles, 

and the starting point is to stop ignoring the returned promise by `save()`  If you save parallel, you may change the `for` loop to `.map()` and handle the result with `Promise.all()`. 

Ref: https://www.techiediaries.com/promise-all-map-async-await-example/

Or turn into serial saving by adding await in front of `save`.